homeassistant/components/iaqualink/config_flow.py

```diff
@@ -32,16 +36,20 @@ async def async_step_user(self, user_input: dict[str, Any] | None = None):
             password = user_input[CONF_PASSWORD]
 
             try:
-                aqualink = AqualinkClient(username, password)
-                await aqualink.login()
-                return self.async_create_entry(title=username, data=user_input)
-            except AqualinkLoginException:
+                async with AqualinkClient(username, password):
+                    return self.async_create_entry(title=username, data=user_input)
```

Comment:
Move code out of the try block that isn't expected to raise.  Likely into an else block